File: API/src/main/java/kr/astar/onair/api/soopliv/soop/SoopPacket.java
package kr.astar.onair.api.soopliv.soop;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class SoopPacket {
    private final String command;
    private final List<String> dataList;
    private final LocalDateTime receivedTime = LocalDateTime.now();

    public SoopPacket(String[] args) {
        this.dataList = new ArrayList<>(Arrays.asList(args));
        String cmd = dataList.remove(0);
        this.command = cmd.substring(0, 4);
    }

    public String getCommand() {
        return command;
    }

    public List<String> getDataList() {
        return dataList;
    }

    public LocalDateTime getReceivedTime() {
        return receivedTime;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        sb.append("Command: ").append(command).append("\n");
        sb.append("Data: ");
        for (String d : dataList) {
            sb.append(d).append(" ");
        }
        return sb.toString();
    }
}

File: src/main/kotlin/kr/apo2073/onAir/soop/SoopPacket.kt
package kr.apo2073.onAir.soop

import java.time.LocalDateTime

class SoopPacket(args: List<String>) {
    val command: String
    val dataList: List<String>
    val receivedTime: LocalDateTime = LocalDateTime.now()

    init {
        require(args.isNotEmpty()) { "args cannot be empty" }
        command = args.first().take(4)
        dataList = args.drop(1)
    }

    override fun toString(): String {
        return buildString {
            append("Command: ").append(command).append("\n")
            append("Data: ").append(dataList.joinToString(" "))
        }
    }
}
